lib/utils/configReader.js

```diff
@@ -5,81 +5,83 @@ var path = require('path')
 var extend = require('lodash/object/extend')
 var defaults = require('lodash/object/defaults')
 
-var allowedReporters = ['logstash', 'trace']
-
-function isReporterValid (reporterType) {
-  var index = allowedReporters
-    .indexOf(reporterType)
+function ConfigReader (config) {
+  this.parameterConfig = config
+}
 
-  return index > -1
+ConfigReader.prototype._getEnvVarConfig = function () {
+  return {
+    collectInterval: process.env.TRACE_COLLECT_INTERVAL,
+    initialSampleRate: process.env.TRACE_INITIAL_SAMPLE_RATE,
+    collectorApi: process.env.TRACE_COLLECTOR_API_URL,
+    appName: process.env.TRACE_APP_NAME,
+    configPath: process.env.TRACE_CONFIG_PATH,
+    apiKey: process.env.TRACE_API_KEY
+  }
 }
 
-function ConfigReader (collectorConfig) {
-  this.collectorConfig = collectorConfig
+ConfigReader.prototype._getDefaultConfig = function () {
+  return require('../config')
 }
 
-ConfigReader.prototype._require = function (filePath) {
-  return require(filePath)
+ConfigReader.prototype._readConfigFile = function (file) {
+  return require(file)
 }
 
-ConfigReader.prototype._initConfig = function () {
-  var config
-  try {
-    var configPath = this.collectorConfig.configPath ||
-      path.join(__dirname, '../../../../../', 'trace.config.js')
-    var configToExtend = this._require(configPath)
-    config = extend({}, configToExtend)
-  } catch (ex) {
-    throw new Error('Invalid trace.config.js configuration file')
+ConfigReader.prototype._getFileConfig = function (file) {
+  if(file) {
+    try {
+      return this._readConfigFile(path.resolve(file))
+    } catch (ex) {
+      if (ex.code === 'MODULE_NOT_FOUND') {
```

Comment:
don't throw an error if the config file is not found, just print a warning. This still can be annoying, because with the current default config, it is always looked up, so this warning is always printed, if the file is does not exist.